### PR TITLE
Miscellaneous clean up to finish java docs, refactoring to make the c…

### PR DESCRIPTION
**Pull request by mmschaefer**
…ode more testable and initial test for a Part01Controller.java

### CODE REVIEW

File: src-test/org/etools/j1939_84/controllers/part1/Part01ControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
+/**
+ * Unit test for Part01ControllerTest
```
Comment (gumbelmj) [inline src-test/org/etools/j1939_84/controllers/part1/Part01ControllerTest.java]: This header needs to be the copyright

File: src/org/etools/j1939_84/bus/j1939/J1939TP.java [hunk 2]
```diff
@@ -26,15 +26,18 @@
     private static final int CM_EndOfMessageACK = 19;
 
     private static final int CM_RTS = 16;
-    final static public int DT = 0xEB00;
-    static final private Map<Integer, String> errors = new HashMap<>();
+    // Most number of packet that can be sent in one CTS
+    private static final int countMax = 0xFF;
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/bus/j1939/J1939TP.java]: Constants need to be UPPER_CASE

File: src/org/etools/j1939_84/controllers/OverallController.java [hunk 3]
```diff
@@ -50,15 +51,28 @@ public OverallController() {
...
+     * @param part3Controller          the {@link Part03Controller} for Part1
+     * @param part4Controller          the {@link Part04Controller} for Part1
+     * @param part5Controller          the {@link Part05Controller} for Part1
+     * @param part6Controller          the {@link Part06Controller} for Part1
+     * @param part7Controller          the {@link Part07Controller} for Part1
+     * @param part8Controller          the {@link Part08Controller} for Part1
+     * @param part9Controller          the {@link Part09Controller} for Part1
+     * @param part10Controller         the {@link Part10Controller} for Part1
+     * @param part11Controller         the {@link Part11Controller} for Part1
+     * @param part12Controller         the {@link Part12Controller} for Part1
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/controllers/OverallController.java]: Copy paste error.

File: src/org/etools/j1939_84/controllers/OverallController.java [hunk 4]
```diff
@@ -50,15 +51,28 @@ public OverallController() {
      *                                 the date and time
      * @param vehicleInformationModule the {@link VehicleInformationModule} used to
      *                                 gather information about the vehicle
-     * @param part1Controller          the {@link Controller} for Part1
+     * @param partResultFactory        the {@link PartResultFactory} for Part1
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/controllers/OverallController.java]: This factory is used for all the parts.

File: src-test/org/etools/j1939_84/controllers/part1/Part01ControllerTest.java (hunk 1 above)
Comment (mmschaefer) [reply src-test/org/etools/j1939_84/controllers/part1/Part01ControllerTest.java]: Done

File: src/org/etools/j1939_84/bus/j1939/J1939TP.java (hunk 2 above)
Comment (mmschaefer) [reply src/org/etools/j1939_84/bus/j1939/J1939TP.java]: Done

File: src/org/etools/j1939_84/controllers/OverallController.java (hunk 3 above)
Comment (mmschaefer) [reply src/org/etools/j1939_84/controllers/OverallController.java]: Done

File: src/org/etools/j1939_84/controllers/OverallController.java (hunk 4 above)
Comment (mmschaefer) [reply src/org/etools/j1939_84/controllers/OverallController.java]: Done

**Pull request merged**